contributors/design-proposals/federated-statefulsets.md

```diff
@@ -0,0 +1,491 @@
+
+The details following extend the example yaml above, with statefulset name _store_, service name
+_etcd_, federation name _myfederation_, domain name as _example.com_, and assume the registered
+cluster names being _c1, c2 and c3_.
+
+Peer pods can discover the stateful pods locally by querying the governing service domain
+```etcd.mynamespace```, which will by default be expanded to a local CNAME of
+```etcd.mynamespace.svc.example.com``` (assuming that the default domain name of the individual
+clusters is overridden with example.com). The CNAME could as we be queried directly. This will
+resolve to the SRV records local to the clusters stateful replicas for each clusters, something
+like listed further. The name resolution is exactly as it happens for the local statefulsets in
+k8s today. Additionally note that the federation name _myfederation_ has explicitly been dropped
+in this example to demonstrate a local query. This is also how the in cluster kube-dns works. It
+parses and considers the service name query to be local or federated global based on the presence
+of valid federation name segment in the queried string. [Refer here](https://github.com/kubernetes/dns/blob/master/pkg/dns/dns.go#L588).
+
+The CNAME query will return SRV records as below.
+
+within cluster c1
+```
+store-c1-0.etcd.mynamespace.svc.example.com
+store-c1-1.etcd.mynamespace.svc.example.com
+store-c1-2.etcd.mynamespace.svc.example.com
+store-c1-3.etcd.mynamespace.svc.example.com
+```
+within cluster c2
+```
+store-c2-0.etcd.mynamespace.svc.example.com
+store-c2-1.etcd.mynamespace.svc.example.com
+store-c2-2.etcd.mynamespace.svc.example.com
+store-c2-3.etcd.mynamespace.svc.example.com
+```
+within cluster c3
+```
+store-c3-0.etcd.mynamespace.svc.example.com
+store-c3-1.etcd.mynamespace.svc.example.com
+store-c3-2.etcd.mynamespace.svc.example.com
+```
+The SRV record will eventually map to an A record with a valid replica pod in-cluster IP if
+the replica is up and running.
+
+Additional pointer is that, if the given cluster is properly added to federation, this same
+headless service and related resolved names can be queried including the federation name in
+query too. For example the service CNAME record ```etcd.mynamespace.myfederation.svc.example.com```
+will also result in similar local names as elaborated above, especially because at this point
+the headless service, even though federated, will not get any dns names updated into the
+global (federated) dns server. The results will however be different, if the global identities
+are also requested by the user as elaborated in the next section.
+
+### Instance identity and discovery (global identities)
+
+The above is fine for the case of multiple local quora (one in each cluster), but these
+identities cannot work across the clusters, as the local pod IPs cannot be reached across
+the clusters.
+
+The proposal to make it work across the clusters is to create an additional service of type
```

Comment:
I have put in the details of these additional LB service handling in the section **Handling additional LB services**.